caddy.go

```diff
@@ -180,15 +181,31 @@ func (i *Instance) ShutdownCallbacks() []error {
 // Restart replaces the servers in i with new servers created from
 // executing the newCaddyfile. Upon success, it returns the new
 // instance to replace i. Upon failure, i will not be replaced.
-func (i *Instance) Restart(newCaddyfile Input) (*Instance, error) {
+func (i *Instance) Restart(newCaddyfile Input) (inst *Instance, err error) {
```

Comment:
Rather than a named return in this instance (sorry, I had to pun it), let's just declare `var` before the `defer` below.